components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractLocalApplicationAuthenticator.java

```diff
@@ -69,14 +73,24 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
                     processAuthenticationResponse(request, response, context);
                     if (this instanceof LocalApplicationAuthenticator && !context.getSequenceConfig()
                             .getApplicationConfig().isSaaSApp()) {
-                        processWithNonSaaSApp(context);
+                        validateNonSaasAppLogin(context);
                     }
                     request.setAttribute(FrameworkConstants.REQ_ATTR_HANDLED, true);
                     context.setProperty(FrameworkConstants.LAST_FAILED_AUTHENTICATOR, null);
                     fireEvent(context, IdentityEventConstants.Event.POST_AUTHENTICATION, true);
                     return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
                 } catch (AuthenticationFailedException e) {
-                    if (checkUserAccountStatus(response, context)) {
+                    if (isAccountLocked(context)) {
+                        String retryPage = ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL();
```

Comment:
Can you move these to a separate protected method? the code here should look like,
if (isAccountLocked(context)) {
    String redirectURL = getRedirectUrlOnAccountLock(<params>);
    response.sendRedirect(redirectUrl);
}

This way it'd be much clear.